Core/src/org/sleuthkit/autopsy/mainui/datamodel/MainDAO.java

```diff
@@ -42,31 +169,192 @@ public synchronized static MainDAO getInstance() {
+                    } catch (Throwable ex) {
+                        // firewall exception
+                        logger.log(Level.WARNING, "An exception occurred while handling tree event timeouts", ex);
+                    }
+                },
+                WATCH_RESOLUTION_MILLIS,
+                WATCH_RESOLUTION_MILLIS,
+                TimeUnit.MILLISECONDS);
+    }
+
+    /**
+     * Unregisters listeners from autopsy event publishers.
+     */
+    void unregister() {
+        IngestManager.getInstance().removeIngestModuleEventListener(INGEST_MODULE_EVENTS, ingestModuleEventListener);
+        IngestManager.getInstance().removeIngestJobEventListener(INGEST_JOB_EVENTS, ingestJobEventListener);
+        Case.removePropertyChangeListener(caseEventListener);
+        UserPreferences.removeChangeListener(userPreferenceListener);
+    }
+
     public DataArtifactDAO getDataArtifactsDAO() {
         return dataArtifactDAO;
     }
-    
+
     public AnalysisResultDAO getAnalysisResultDAO() {
         return analysisResultDAO;
     }
 
     public ViewsDAO getViewsDAO() {
         return viewsDAO;
     }
-    
+
     public FileSystemDAO getFileSystemDAO() {
         return fileSystemDAO;
     }
-    
+
     public TagsDAO getTagsDAO() {
         return tagsDAO;
     }
-    
+
     public OsAccountsDAO getOsAccountsDAO() {
         return osAccountsDAO;
     }
-    
+
     public CommAccountsDAO getCommAccountsDAO() {
         return commAccountsDAO;
     }
+
+    public PropertyChangeManager getResultEventsManager() {
+        return this.resultEventsManager;
+    }
+
+    public PropertyChangeManager getTreeEventsManager() {
+        return treeEventsManager;
+    }
+
+    @Override
+    void clearCaches() {
+        allDAOs.forEach((subDAO) -> subDAO.clearCaches());
+    }
+
+    @Override
+    List<DAOEvent> processEvent(PropertyChangeEvent evt) {
+        return allDAOs.stream()
+                .map(subDAO -> subDAO.processEvent(evt))
+                .flatMap(evts -> evts == null ? Stream.empty() : evts.stream())
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    List<TreeEvent> shouldRefreshTree() {
+        return allDAOs.stream()
+                .map((subDAO) -> subDAO.shouldRefreshTree())
+                .flatMap(evts -> evts == null ? Stream.empty() : evts.stream())
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    Collection<? extends DAOEvent> flushEvents() {
+        List<Collection<? extends DAOEvent>> daoStreamEvts = allDAOs.stream()
+                .map((subDAO) -> subDAO.flushEvents())
+                .collect(Collectors.toList());
+
+        daoStreamEvts.add(eventBatcher.flushEvents());
+
+        return daoStreamEvts.stream()
+                .flatMap(evts -> evts == null ? Stream.empty() : evts.stream())
+                .collect(Collectors.toList());
+    }
+
+    private void handleEvent(PropertyChangeEvent evt, boolean immediateAction) {
```

Comment:
Add comment. Clarify that 'immediateAction' is just about the result events.  Tree events are always immediate.